failing docker build

### katana version:
latest / dev

### Current Behavior:

```bash
 > [linux/amd64 builder 6/6] RUN go build ./cmd/katana:
3.179 # github.com/smacker/go-tree-sitter
3.179 /go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20220628134258-ac06e95cfa11/iter.go:17:18: undefined: Node
3.179 /go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20220628134258-ac06e95cfa11/iter.go:21:21: undefined: Node
3.179 /go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20220628134258-ac06e95cfa11/iter.go:25:20: undefined: Node
3.179 /go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20220628134258-ac06e95cfa11/iter.go:30:26: undefined: Node
3.179 /go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20220628134258-ac06e95cfa11/iter.go:34:20: undefined: Node
3.179 /go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20220628134258-ac06e95cfa11/iter.go:38:32: undefined: Node
3.179 /go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20220628134258-ac06e95cfa11/iter.go:43:9: undefined: Node
3.179 /go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20220628134258-ac06e95cfa11/iter.go:46:18: undefined: Node
3.179 /go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20220628134258-ac06e95cfa11/iter.go:68:40: undefined: Node
------
Dockerfile:6
--------------------
   4 |     COPY . /app
   5 |     RUN go mod download
   6 | >>> RUN go build ./cmd/katana
   7 |     
   8 |     FROM alpine:3.18.2
--------------------
ERROR: failed to solve: process "/bin/sh -c go build ./cmd/katana" did not complete successfully: exit code: 1
Error: buildx failed with: ERROR: failed to solve: process "/bin/sh -c go build ./cmd/katana" did not complete successfully: exit code: 1
```

### Expected Behavior:
passing docker build

### Steps To Reproduce:
```
sudo docker build -t test .
```


### Anything else:
<!-- Links? References? Screnshots? Anything that will give us more context about the issue that you are encountering! -->

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,5 @@
 FROM golang:1.20.6-alpine AS builder
-RUN apk add --no-cache git
+RUN apk add --no-cache git gcc musl-dev
 WORKDIR /app
 COPY . /app
 RUN go mod download

